src/bgen/Generator.cs

```diff
@@ -5630,6 +5631,13 @@ string GetAssemblyName ()
 
 	public void Generate (Type type)
 	{
+
+		// check if the type has been marked as a type that will be generated by the new code generator, if that
+		// is the case, bgen will ignore it allowing the rgen code generator add the type to the final assembly
+		bool is_rgen_type = AttributeManager.HasAttribute<BindingTypeAttribute> (type);
```

Comment:
Will this predicate be used in other places and if so, does it justify putting it in its own method?

Reply:
Not really, we could use it in the if, not a big diff. We only use it to skip the generation.